diff --git a/.gitignore b/.gitignore
index b7b6843..d8c4adb 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,7 @@
 ./.vscode/**
-.vscode
\ No newline at end of file
+.vscode
+
+path_to_the_north_texture
+path_to_the_south_texture
+path_to_the_west_texture
+path_to_the_east_texture
\ No newline at end of file
diff --git a/Makefile b/Makefile
index 38583e3..771e121 100644
--- a/Makefile
+++ b/Makefile
@@ -13,9 +13,7 @@ RAW_C		=	main.c \
 				get_next_line.c \
 				parser_load_map.c \
 				parser_eval_map.c \
-				parser_map_integrity.c \
-				parser_eval_map_char.c \
-				parser_eval_map_rules.c
+				parser_eval_assets.c
 
 SRC			=	$(addprefix $(SRC_D)/,$(RAW_C))
 OBJ			=	$(addprefix $(OBJ_D)/,$(RAW_C:.c=.o))
@@ -42,6 +40,6 @@ fclean:	clean
 re: fclean all
 
 ./libft/libft.a:
-	@make -C ./libft
+	@make -C ./libft --no-print-directory
 
 bonus:
diff --git a/cub3D b/cub3D
deleted file mode 100755
index fa36dbb..0000000
Binary files a/cub3D and /dev/null differ
diff --git a/includes/cub3d.h b/includes/cub3d.h
index 09388c1..855c3f9 100644
--- a/includes/cub3d.h
+++ b/includes/cub3d.h
@@ -3,10 +3,10 @@
 /*                                                        :::      ::::::::   */
 /*   cub3d.h                                            :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: hde-camp <hde-camp@student.42sp.org.br>    +#+  +:+       +#+        */
+/*   By: dpiza <dpiza@student.42sp.org.br>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/04/22 16:38:46 by hde-camp          #+#    #+#             */
-/*   Updated: 2022/04/28 15:32:59 by hde-camp         ###   ########.fr       */
+/*   Updated: 2022/04/25 17:12:56 by dpiza            ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
@@ -26,35 +26,31 @@
 typedef struct s_map	t_map;
 
 enum e_map_status{
-	OK = 0,
-	NO_MAP = 0b00000001,
-	PERIMETER_ERROR = 0b00000010,
-	INVALID_CHARACTER = 0b00000100,
-	GARBAGE_LINES = 0b00001000,
-	MAP_TOO_SHORT = 0b00010000,
-	MAP_NOT_WALLED = 0b00100000
+    OK = 0,
+    NO_MAP = 1,
+    PERIMETER_ERROR = 2,
+    INVALID_CHARACTER = 3,
+    GARBAGE_LINES = 4,
+    MAP_TOO_SHORT = 5
 };
 
-struct	s_map
+struct    s_map
 {
-	char				**lines;
-	char				*no_path;
-	char				*so_path;
-	char				*we_path;
-	char				*ea_path;
-	unsigned char		floor_c[3];
-	unsigned char		ceiling_c[3];
-	int					is_ok;
-	enum e_map_status	status;
+    char                **lines;
+    char                *no_path;
+    char                *so_path;
+    char                *we_path;
+    char                *ea_path;
+    unsigned char        floor_c[3];
+    unsigned char        ceiling_c[3];
+    int                    is_ok;
+    enum e_map_status    status;
 };
 
 char	*get_next_line(int fd);
 t_map	*load_map(const char *path);
 void	eval_map(t_map	*map);
-void	map_integrity(t_map *map);
-int		is_map_allowed_character(char c);
-int		is_map_first_char(char c);
-void	eval_map_rules(t_map *map, char **first_line, char **last_line);
+void	eval_assets(t_map *map);
 
 # define OPEN_MAX 256
 
diff --git a/includes/libft.h b/includes/libft.h
index 680d928..52e3d38 100644
--- a/includes/libft.h
+++ b/includes/libft.h
@@ -3,10 +3,10 @@
 /*                                                        :::      ::::::::   */
 /*   libft.h                                            :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: hde-camp <hde-camp@student.42sp.org.br>    +#+  +:+       +#+        */
+/*   By: dpiza <dpiza@student.42sp.org.br>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/01/31 18:50:52 by hde-camp          #+#    #+#             */
-/*   Updated: 2022/02/05 12:53:40 by hde-camp         ###   ########.fr       */
+/*   Updated: 2022/04/25 17:42:19 by dpiza            ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
@@ -149,4 +149,7 @@ void		ft_dl_removeone(t_dl_list **lst, void (*del)(void*));
  *  if no character is found, return zero.
  */
 int			ft_chrinstr(char c, const char *str);
+
+void		ft_split_free(char ***str_array);
+
 #endif
diff --git a/libft/Makefile b/libft/Makefile
index 6c2f402..e28e969 100644
--- a/libft/Makefile
+++ b/libft/Makefile
@@ -63,7 +63,8 @@ RAW			=	ft_atoi.c \
 				ft_dl_destroyall.c \
 				ft_dl_removeone.c \
 				ft_dl_getfirst.c \
-				ft_chrinstr.c
+				ft_chrinstr.c \
+				ft_split_free.c
 
 SRC		=	$(addprefix $(SRC_DIR)/,$(RAW))
 OBJS	=	$(addprefix $(OBJ_DIR)/,$(RAW:.c=.o))
diff --git a/libft/libft.h b/libft/libft.h
index 680d928..4af2b2a 100644
--- a/libft/libft.h
+++ b/libft/libft.h
@@ -3,10 +3,10 @@
 /*                                                        :::      ::::::::   */
 /*   libft.h                                            :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: hde-camp <hde-camp@student.42sp.org.br>    +#+  +:+       +#+        */
+/*   By: dpiza <dpiza@student.42sp.org.br>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/01/31 18:50:52 by hde-camp          #+#    #+#             */
-/*   Updated: 2022/02/05 12:53:40 by hde-camp         ###   ########.fr       */
+/*   Updated: 2022/04/25 17:41:37 by dpiza            ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
@@ -149,4 +149,7 @@ void		ft_dl_removeone(t_dl_list **lst, void (*del)(void*));
  *  if no character is found, return zero.
  */
 int			ft_chrinstr(char c, const char *str);
+
+void		ft_split_free(char ***str_array);
+
 #endif
diff --git a/src/parser_eval_map_char.c b/libft/src/ft_split_free.c
similarity index 53%
rename from src/parser_eval_map_char.c
rename to libft/src/ft_split_free.c
index 218bea5..b62a715 100644
--- a/src/parser_eval_map_char.c
+++ b/libft/src/ft_split_free.c
@@ -1,45 +1,36 @@
 /* ************************************************************************** */
 /*                                                                            */
 /*                                                        :::      ::::::::   */
-/*   parser_map_char_eval.c                             :+:      :+:    :+:   */
+/*   ft_split_free.c                                    :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: hde-camp <hde-camp@student.42sp.org.br>    +#+  +:+       +#+        */
+/*   By: dpiza <dpiza@student.42sp.org.br>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
-/*   Created: 2022/04/25 17:40:31 by hde-camp          #+#    #+#             */
-/*   Updated: 2022/04/25 17:41:46 by hde-camp         ###   ########.fr       */
+/*   Created: 2021/04/19 10:57:02 by rkochhan          #+#    #+#             */
+/*   Updated: 2022/04/25 17:41:41 by dpiza            ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
-#include <cub3d.h>
+#include "libft.h"
 
-int	is_map_allowed_character(char c)
+void	ft_strdel(char **str)
 {
-	char	*allowed;
-	int		i;
-
-	allowed = "10NSEW ";
-	i = 0;
-	while (i < 8)
+	if (str && *str)
 	{
-		if (allowed[i] == c)
-			return (1);
-		i++;
+		free(*str);
+		*str = NULL;
 	}
-	return (0);
 }
 
-int	is_map_first_char(char c)
+void	ft_split_free(char ***str_array)
 {
-	char	*allowed;
-	int		i;
+	size_t	i;
 
-	allowed = " 10";
 	i = 0;
-	while (i < 3)
+	while ((*str_array)[i])
 	{
-		if (allowed[i] == c)
-			return (1);
+		ft_strdel(&((*str_array)[i]));
 		i++;
 	}
-	return (0);
-}
\ No newline at end of file
+	free(*str_array);
+	*str_array = NULL;
+}
diff --git a/maps/map_with_spaces_and_role.cub b/maps/map_with_spaces_and_role.cub
deleted file mode 100644
index f5097a5..0000000
--- a/maps/map_with_spaces_and_role.cub
+++ /dev/null
@@ -1,23 +0,0 @@
-NO ./path_to_the_north_texture
-SO ./path_to_the_south_texture
-WE ./path_to_the_west_texture
-EA ./path_to_the_east_texture
-
-F 220,100,0
-C 225,30,0
-
-11111111111111111111111113333333X
-1 00000000000000000000001
-1001000000110000000000001
-1011000001110000000000001
-100100000000000000000000111111111
-111111111011000001110000000000001
-100000000011000001110111111111111
-11110111111111011100000010001
-11110111111111011101010010001
-11000000110101011100000010001
-10000000000000001100000010001
-10000000000000001101010010001
-11000001110101011111011110N0111
-11110111 1110101 101111010001
-11111111 1111111 111111111111
\ No newline at end of file
diff --git a/obj/get_next_line.o b/obj/get_next_line.o
deleted file mode 100644
index ec901ea..0000000
Binary files a/obj/get_next_line.o and /dev/null differ
diff --git a/obj/main.o b/obj/main.o
deleted file mode 100644
index 7591f78..0000000
Binary files a/obj/main.o and /dev/null differ
diff --git a/obj/parser_eval_map.o b/obj/parser_eval_map.o
deleted file mode 100644
index 0a09912..0000000
Binary files a/obj/parser_eval_map.o and /dev/null differ
diff --git a/obj/parser_eval_map_char.o b/obj/parser_eval_map_char.o
deleted file mode 100644
index 2182e64..0000000
Binary files a/obj/parser_eval_map_char.o and /dev/null differ
diff --git a/obj/parser_eval_map_rules.o b/obj/parser_eval_map_rules.o
deleted file mode 100644
index e69817f..0000000
Binary files a/obj/parser_eval_map_rules.o and /dev/null differ
diff --git a/obj/parser_load_map.o b/obj/parser_load_map.o
deleted file mode 100644
index a5528c6..0000000
Binary files a/obj/parser_load_map.o and /dev/null differ
diff --git a/obj/parser_map_integrity.o b/obj/parser_map_integrity.o
deleted file mode 100644
index 68a0716..0000000
Binary files a/obj/parser_map_integrity.o and /dev/null differ
diff --git a/src/main.c b/src/main.c
index ef60577..db822ae 100644
--- a/src/main.c
+++ b/src/main.c
@@ -3,10 +3,10 @@
 /*                                                        :::      ::::::::   */
 /*   main.c                                             :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: hde-camp <hde-camp@student.42sp.org.br>    +#+  +:+       +#+        */
+/*   By: dpiza <dpiza@student.42sp.org.br>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/04/22 16:48:44 by hde-camp          #+#    #+#             */
-/*   Updated: 2022/04/28 12:22:59 by hde-camp         ###   ########.fr       */
+/*   Updated: 2022/04/25 17:09:36 by dpiza            ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
@@ -18,8 +18,9 @@ int	main(int argc, char *argv[])
 
 	(void)argc;
 	(void)argv;
-	map = load_map("./maps/map_with_spaces_and_role.cub");
+	map = load_map("./maps/map_with_spaces.cub");
 	eval_map(map);
+	eval_assets(map);
 	(void)map;
 	return (0);
 }
diff --git a/src/parser_eval_assets.c b/src/parser_eval_assets.c
new file mode 100644
index 0000000..f4f318a
--- /dev/null
+++ b/src/parser_eval_assets.c
@@ -0,0 +1,127 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parser_eval_assets.c                               :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: dpiza <dpiza@student.42sp.org.br>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2022/04/24 21:49:33 by dpiza             #+#    #+#             */
+/*   Updated: 2022/04/27 18:26:21 by dpiza            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <cub3d.h>
+
+static void	fill_texture_path(t_map *map, char *line)
+{
+	if (line[0] == 'N')
+		map->no_path = ft_strdup(ft_strchr(line, '.'));
+	else if (line[0] == 'S')
+		map->so_path = ft_strdup(ft_strchr(line, '.'));
+	else if (line[0] == 'W')
+		map->we_path = ft_strdup(ft_strchr(line, '.'));
+	else if (line[0] == 'E')
+		map->ea_path = ft_strdup(ft_strchr(line, '.'));
+}
+
+static int	is_valid_texture(char *line)
+{
+	int			i;
+	static char	*path[4] = {"NO", "SO", "WE", "EA"};
+
+	i = 0;
+	while (path[i] && ft_strncmp(line, path[i], 2) != 0)
+		i++;
+	if (i > 3)
+		return (0);
+	line += 2;
+	while (ft_isspace(*line))
+		line++;
+	if (*line != '.')
+		return (0);
+	line[ft_strlen(line) - 1] = '\0';
+	if (open(ft_strchr(line, '.'), O_RDONLY) > 0)
+		return (1);
+	return (0);
+}
+
+static void	fill_color(t_map *map, char *line)
+{
+	char	**color;
+	char	plane;
+	int		i;
+
+	plane = line[0];
+	line++;
+	while (ft_isspace(*line))
+		line++;
+	color = ft_split(line, ',');
+	i = 0;
+	while (color[i])
+	{
+		if (plane == 'F')
+			map->floor_c[i] = ft_atoi(color[i]);
+		else
+			map->ceiling_c[i] = ft_atoi(color[i]);
+		i++;
+	}
+	ft_split_free(&color);
+}
+
+static int	is_valid_color(char *line)
+{
+	char	**color;
+	int		i;
+
+	if (ft_strncmp(line, "F", 1) && ft_strncmp(line, "C", 1))
+		return (0);
+	line++;
+	while (ft_isspace(*line))
+		line++;
+	if (!ft_isdigit(*line))
+		return (0);
+	line[ft_strlen(line) - 1] = '\0';
+	color = ft_split(line, ',');
+	i = 0;
+	while (color[i] && ft_atoi(color[i]) >= 0 && ft_atoi(color[i]) < 256)
+		i++;
+	ft_split_free(&color);
+	if (i > 2)
+		return (1);
+	return (0);
+}
+
+void	print_struct(t_map *map)
+{
+	printf("no_path: %s\nso_path: %s\nwe_path: %s\nea_path: %s\nfloor_c[0]: %d\nfloor_c[1]: %d\nfloor_c[2]: %d\nceiling_c[0]: %d\nceiling_c[1]: %d\nceiling_c[2]: %d\n",
+		map->no_path, map->so_path, map->we_path, map->ea_path,
+		map->floor_c[0], map->floor_c[1], map->floor_c[2],
+		map->ceiling_c[0], map->ceiling_c[1], map->ceiling_c[2]);
+}
+
+void	free_struct(t_map *map)
+{
+	free(map->no_path);
+	free(map->so_path);
+	free(map->we_path);
+	free(map->ea_path);
+	ft_split_free(&map->lines);
+	free(map);
+}
+
+void	eval_assets(t_map *map)
+{
+	char	**map_lines;
+
+	map_lines = map->lines;
+	while (*map_lines)
+	{
+		if (is_valid_texture(*map_lines))
+			fill_texture_path(map, *map_lines);
+		if (is_valid_color(*map_lines))
+			fill_color(map, *map_lines);
+		map_lines++;
+	}
+	print_struct(map);
+	free_struct(map);
+}
diff --git a/src/parser_eval_map.c b/src/parser_eval_map.c
index e6b5bf5..8e9bc8f 100644
--- a/src/parser_eval_map.c
+++ b/src/parser_eval_map.c
@@ -6,13 +6,82 @@
 /*   By: hde-camp <hde-camp@student.42sp.org.br>    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/04/22 23:04:16 by hde-camp          #+#    #+#             */
-/*   Updated: 2022/04/27 09:53:50 by hde-camp         ###   ########.fr       */
+/*   Updated: 2022/04/23 00:31:06 by hde-camp         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include <cub3d.h>
 
+//static	int	is_map_allowed_character(char c)
+//{
+//	char	*allowed;
+//	int		i;
+//
+//	allowed = "10NSEW";
+//	i = 0;
+//	while (i < 8)
+//	{
+//		if (allowed[i] == c)
+//			return (1);
+//		i++;
+//	}
+//	return (0);
+//}
+
+static	int	is_map_first_char(char c)
+{
+	char	*allowed;
+	int		i;
+
+	allowed = " 10";
+	i = 0;
+	while (i < 3)
+	{
+		if (allowed[i] == c)
+			return (1);
+		i++;
+	}
+	return (0);
+}
+static	int	is_map_line(char *line)
+{
+	if (line)
+	{
+		while (ft_isspace(*line))
+			line++;
+		if (line)
+		{
+			if (is_map_first_char(*line))
+				return (1);
+			else
+				return (0);
+		}
+		return (0);
+	}
+	return (0);
+}
+
+static	char **get_first_map_line(t_map	*map)
+{
+	char **map_lines;
+
+	map_lines = map->lines;
+	while (*map_lines)
+	{
+		if (is_map_line(*map_lines))
+			return (map_lines);
+		map_lines++;
+	}
+	return (NULL);
+}
+
 void	eval_map(t_map	*map)
 {
-	map_integrity(map);
+	char	**first_map_line;
+
+	first_map_line = get_first_map_line(map);
+	if (first_map_line)
+	{
+		printf("found first line\n");
+	}
 }
diff --git a/src/parser_eval_map_rules.c b/src/parser_eval_map_rules.c
deleted file mode 100644
index cebed29..0000000
--- a/src/parser_eval_map_rules.c
+++ /dev/null
@@ -1,208 +0,0 @@
-/* ************************************************************************** */
-/*                                                                            */
-/*                                                        :::      ::::::::   */
-/*   parser_eval_map_rules.c                            :+:      :+:    :+:   */
-/*                                                    +:+ +:+         +:+     */
-/*   By: hde-camp <hde-camp@student.42sp.org.br>    +#+  +:+       +#+        */
-/*                                                +#+#+#+#+#+   +#+           */
-/*   Created: 2022/04/26 13:41:02 by hde-camp          #+#    #+#             */
-/*   Updated: 2022/04/28 15:47:48 by hde-camp         ###   ########.fr       */
-/*                                                                            */
-/* ************************************************************************** */
-
-#include <cub3d.h>
-
-typedef struct s_strmap t_strmap;
-
-struct s_strmap{
-	char	*map;
-	int		columns;
-	int		lines;
-};
-
-int		get_map_max_len(char **first_line, int n_lines)
-{
-	int	local_len;
-	int	max_len;
-	int	line_n;
-
-	line_n = 0;
-	max_len = 0;
-	while (line_n < n_lines)
-	{
-		local_len = ft_strlen(first_line[line_n]);
-		if (first_line[line_n][local_len - 1] == '\n')
-			local_len -= 1;
-		line_n++;
-		if (local_len > max_len)
-			max_len = local_len;
-	}
-	return (max_len);
-}
-
-
-void	print_strmap(char *map,int line_size)
-{
-	int	l;
-	int	map_len;
-
-	l = 0;
-	map_len = ft_strlen(map);
-	while (l < map_len)
-	{
-		write(1, map + l, line_size);
-		write(1, "\n", 1);
-		l += line_size;
-	}
-}
-
-void	convert_into_string(char **dest, char **first_line, int n_lines, int line_size)
-{
-	int	line_count;
-	int	char_count;
-	int	local_len;
-
-	line_count = 0;
-	while (line_count < n_lines)
-	{
-		local_len = ft_strlen(first_line[line_count]) - 1; // there is a -1 because each line has its own \n.
-		char_count = 0;
-		while (char_count < local_len)
-		{
-			(*dest)[(line_count * line_size) + char_count] = first_line[line_count][char_count];
-			char_count++;
-		}
-		while (char_count < line_size)
-		{
-			(*dest)[(line_count * line_size) + char_count] = ' ';
-			char_count++;
-		}
-		line_count++;
-	}
-}
-
-unsigned char	get_char_at(t_strmap *map, int x, int y)
-{
-	int	pos;
-	if (x >= 0 && x < map->columns && y >= 0 && y < map->lines)
-	{
-		pos = x + (y * (map->columns));
-		return (map->map[pos]);
-	}
-	return (0);
-}
-
-int		is_map_edge(t_strmap *map, int x, int y)
-{
-	int	mask;
-
-	mask = 0;
-	mask = x == 0;
-	mask |= x == map->columns - 1;
-	mask |= y == 0;
-	mask |= y == map->lines - 1;
-	return (mask);
-}
-
-int		inner_content_touch_invalid_char(t_strmap *map, int x, int y)
-{
-	unsigned char	boundaries[4];
-	int				invalid_b;
-
-	boundaries[0] = get_char_at(map, x, y - 1);
-	boundaries[1] = get_char_at(map, x, y + 1);
-	boundaries[2] = get_char_at(map, x - 1, y);
-	boundaries[3] = get_char_at(map, x + 1, y);
-	invalid_b = 0;
-	invalid_b |= boundaries[0] == ' ' || boundaries[0] == 0;
-	invalid_b |= boundaries[1] == ' ' || boundaries[1] == 0;
-	invalid_b |= boundaries[2] == ' ' || boundaries[2] == 0;
-	invalid_b |= boundaries[3] == ' ' || boundaries[3] == 0;
-	return (invalid_b);
-}
-
-int		map_is_walled(t_strmap	*map)
-{
-	int	l;
-	int	c;
-	int	is_ok;
-
-	l = 0;
-	is_ok = 1;
-	while (l < map->lines && is_ok)
-	{
-		c = 0;
-		while (c < map->columns && is_ok)
-		{
-			if (is_map_edge(map, c, l))
-			{
-				if (ft_strchr("0NSEW",(int) get_char_at(map, c, l)))
-				{
-					is_ok = 0;
-					break;
-				}
-			}
-			else
-			{
-				if (ft_strchr("0NSEW",get_char_at(map, c, l)))
-				{
-					if (inner_content_touch_invalid_char(map, c, l))
-					{
-						is_ok = 0;
-						break;
-					}
-				}
-			}
-			c++;
-		}
-		l++;
-	}
-	return (is_ok);
-}
-
-void	print_map_error(t_map *map)
-{
-	char *msgs[6];
-	int		i;
-	int		mask;
-
-	msgs[0] = "Error: Map not found.\n";
-	msgs[1] = "Error: Perimeter error.\n";
-	msgs[2] = "Error: Map has invalid character.\n";
-	msgs[3] = "Error: Unecessary lines with garbage.\n";
-	msgs[4] = "Error: Map too short.\n";
-	msgs[5] = "Error: Map not properly walled.\n";
-	i = 0;
-	while (i < 6)
-	{
-		mask = 1 << i;
-		if (mask & map->status)
-			printf("%s", msgs[i]);
-		i++;
-	}
-}
-
-void	eval_map_rules(t_map *map, char **first_line, char **last_line)
-{
-	char		*map_as_str;
-	int			line_n;
-	int			line_s;
-	t_strmap	strmap;
-
-	//get maximum length and padronize map into a long string.
-	(void)map;
-	line_n = last_line - first_line + 1;
-	line_s = get_map_max_len(first_line, line_n);
-	map_as_str = ft_calloc((line_s * line_n), 1); //this is not null terminated!
-	strmap.columns = line_s;
-	strmap.lines = line_n;
-	strmap.map = map_as_str;
-	convert_into_string(&map_as_str, first_line, line_n, line_s);
-	print_strmap(map_as_str, line_s);
-	if (!map_is_walled(&strmap))
-	{
-		map->is_ok = 0;
-		map->status |= MAP_NOT_WALLED;
-	}
-	print_map_error(map);
-}
\ No newline at end of file
diff --git a/src/parser_map_integrity.c b/src/parser_map_integrity.c
deleted file mode 100644
index 38657b3..0000000
--- a/src/parser_map_integrity.c
+++ /dev/null
@@ -1,126 +0,0 @@
-/* ************************************************************************** */
-/*                                                                            */
-/*                                                        :::      ::::::::   */
-/*   parser_map_integrity.c                             :+:      :+:    :+:   */
-/*                                                    +:+ +:+         +:+     */
-/*   By: hde-camp <hde-camp@student.42sp.org.br>    +#+  +:+       +#+        */
-/*                                                +#+#+#+#+#+   +#+           */
-/*   Created: 2022/04/25 14:58:11 by hde-camp          #+#    #+#             */
-/*   Updated: 2022/04/28 15:34:16 by hde-camp         ###   ########.fr       */
-/*                                                                            */
-/* ************************************************************************** */
-
-#include <cub3d.h>
-
-static	int	is_map_line(char *line);
-static	int	is_map_first_line(char *line);
-static	char	**get_map_first_line(t_map	*map);
-static	char	**get_map_last_line(char **first_line);
-static	void	eval_trailing_garbage(t_map *map, char **map_last_line);
-
-void	map_integrity(t_map *map)
-{
-	char	**map_first_line;
-	char	**map_last_line;
-
-	map_first_line = get_map_first_line(map);
-	if (map_first_line)
-	{
-		map_last_line = get_map_last_line(map_first_line);
-		if (map_first_line - map_last_line)
-		{
-			eval_trailing_garbage(map, map_last_line);
-		}
-		else
-		{
-			map->status |= MAP_TOO_SHORT;
-		}
-	}
-	else
-		map->status |= NO_MAP;
-	eval_map_rules(map, map_first_line, map_last_line);
-}
-
-static	void	eval_trailing_garbage(t_map *map, char **map_last_line)
-{
-	int		i;
-	char	*line;
-
-	i = 0;
-	while (map_last_line[i] && map->status == OK)
-	{
-		line = map_last_line[i];
-		while (line)
-		{
-			if (!ft_isspace(*line))
-			{
-				map->status |= GARBAGE_LINES;
-				break;
-			}
-			line++;
-		}
-	}
-}
-
-static	int	is_map_line(char *line)
-{
-	if (line)
-	{
-		while (ft_isspace(*line))
-			line++;
-		if (line)
-		{
-			if (is_map_allowed_character(*line))
-				return (1);
-			else
-				return (0);
-		}
-		return (0);
-	}
-	return (0);
-}
-
-static	int	is_map_first_line(char *line)
-{
-	if (line)
-	{
-		while (ft_isspace(*line))
-			line++;
-		if (line)
-		{
-			if (is_map_first_char(*line))
-				return (1);
-			else
-				return (0);
-		}
-		return (0);
-	}
-	return (0);
-}
-
-static	char **get_map_first_line(t_map	*map)
-{
-	char **map_lines;
-
-	map_lines = map->lines;
-	while (*map_lines)
-	{
-		if (is_map_first_line(*map_lines))
-			return (map_lines);
-		map_lines++;
-	}
-	return (NULL);
-}
-
-static	char **get_map_last_line(char **first_line)
-{
-	char	**last_line;
-
-	last_line = first_line;
-	while (is_map_line(*last_line))
-	{
-		last_line++;
-	}
-	last_line--;
-	return (last_line);
-}
